src/network.js

```diff
@@ -1,11 +1,112 @@
 // @flow
 import invariant from "invariant";
 import axios from "axios";
+import type { $AxiosError, $AxiosXHR, AxiosXHRConfig } from "axios";
 import { log } from "@ledgerhq/logs";
 import { NetworkDown, LedgerAPI5xx, LedgerAPI4xx } from "@ledgerhq/errors";
 import { retry } from "./promise";
 import { getEnv } from "./env";
 
+type Metadata = { startTime: number };
+type ExtendedXHRConfig = AxiosXHRConfig<any> & { metadata?: Metadata };
+
+const MAX_RESPONSE_DATA_SIZE = getEnv("DEBUG_HTTP_RESPONSE_MAX_SIZE");
+
+const limitLogDataSize = (data: any): string | any => {
+  const res = JSON.stringify(data);
+  if (res.length > MAX_RESPONSE_DATA_SIZE)
+    return `${res.substr(0, MAX_RESPONSE_DATA_SIZE)} ...`;
+
+  return data;
+};
+
+export const requestInterceptor = (
+  request: AxiosXHRConfig<any>
+): ExtendedXHRConfig => {
+  const { url, method = "", data } = request;
+  log("network", `${method} ${url}`, { data });
+
+  const req: ExtendedXHRConfig = request;
+
+  req.metadata = {
+    startTime: Date.now(),
+  };
+
+  return req;
+};
+
+export const responseInterceptor = (
+  response: {
+    config: ExtendedXHRConfig,
+  } & $AxiosXHR<any>
+) => {
+  const { url, method = "", metadata = {} } = response.config;
+  const { startTime = 0 } = metadata;
+
+  log(
+    "network-success",
+    `${response.status} ${method} ${url} (${(Date.now() - startTime).toFixed(
+      0
+    )}ms)`,
+    getEnv("DEBUG_HTTP_RESPONSE")
+      ? { data: limitLogDataSize(response.data) }
```

Comment:
i'm not sure why we need to do this here.
i think if DEBUG_HTTP_RESPONSE is on, it never should limit. however if it's off, we always we include the response and possibly limit. but imo we shouldn't "shrunk" it at all, we should just not include it. we shouldn't slice() the data, we will not want partial data, especially we can't automatically know it's "limited" or not.

Reply:
also are you sure this works? I think response.data can be a json object no? how would we detect the size then. (JSON.stringify to be avoided because cost)

Reply:
Yeah i'm not sure how to limit the size of the response logs. :thinking: to do it i did a stringify indeed to return a sliced string in case its too long but its not great.
What we can do though is maybe use something like https://www.npmjs.com/package/object-sizeof to check the response size in Bytes and just filter out the responses that are too long :shrug: 
Do we have a limitation for not wanting to keep all responses in logs?

Reply:
@LFBarreto i would prefer we remove out the idea to limit for now and we'll split out in a future task.